### DISCUSSION TITLE
How to embrace `workbench.tree.expandMode` in TreeItem?

### DISCUSSION THREAD

**Discussion by alefragnani** [microsoft/vscode-discussions#88] [Extension Development QnA] [answered]
Hi,

VS Code has a setting called `workbench.tree.expandMode`, which says _Controls how tree folders  are expanded when clicking folder names. Note that some trees and lists might choose to ignore this setting if it is not applicable._. 

I have a `TreeView` where I add two levels of items. The first one refers to a File and the second one some info inside the file. 

```
  File one
    Item one - one
    Item one - two 
  File two
    Item two - one
    Item two - two
```

Does anyone knows how an extension _might choose to ignore this setting_? Because I don’t have anything special defined on my `Treeview` and it is being ignored 😞  . Or does it work only for _real folders_, instead?

On the other hand, another similar VS Code setting (`workbench.list.openMode`) seems to be properly respected by the extension, with no-code at all.

Thank you

**Comment by isidorn** [comment]
@alexr00 can you maybe help here?

**Comment by alexr00** [comment, marked as answer]
`workbench.tree.expandMode` is ignored in extension provided trees when the element that the user is clicking on has a `command` associated with it.

**Reply by alefragnani** [reply]
Hi @alexr00 ,

Thanks for pointing that out, but it seems something has changed in recent releases, as based on a few tests, this is not working as expected.

My Treeview has no `command` assigned to the `TreeItem` (`TreeItem.command` property), but has commands contributed to context menu via `view/item/context`. I removed them and it still does not work. I also played with the **tree-view-sample** from the **vscode-extensions-sample** repo, and the setting is not respected as well.

Am I missing something?

Thank you

**Reply by alexr00** [reply]
TIL that `workbench.tree.expandMode` is relatively new: https://github.com/microsoft/vscode/issues/95697. 
Respecting `workbench.tree.expandMode` was never implemented for extension trees, but I think it makes sense to respect it. This is not something an extension can change, it must be implemented in VS Code.

**Reply by alefragnani** [reply]
Hi @alexr00!

Based on your discovery, do you need me to create an issue in VS Code repo for such implementation, or would you handle it internally?

Thank you

**Reply by alexr00** [reply]
I created an issue here and have already started the fix :) https://github.com/microsoft/vscode/issues/158159

**Reply by alefragnani** [reply]
Wow, thank you! 🤯
